### DISCUSSION TITLE
Firefly III Data Importer with Docker Compose and Caddy

### DISCUSSION THREAD

**Discussion by lehobhob** [firefly-iii/firefly-iii#9290] [Firefly III]
Does anyone have any experience with installing the Firefly III Data Importer with Docker Compose and Caddy? 
I've followed the instructions from the documentation (https://docs.firefly-iii.org/how-to/data-importer/installation/docker/)) but I'm getting a 502 error. 

Firefly works fine but the data importer doesn't.

I have already set FIREFLY_III_URL in .importer.env to http://ffapp:8080

and
VANITY_URL in .importer.env to http://localhost/

my caddy is the standard:

```
importer.{$MY_DOMAIN} {
reverse_proxy firefly_iii_importer:8081
}

```
I'm out of ideas of where it breaks :(

**Comment by JC5** [comment]
Doesn't that need `http://` in front of it?

**Reply by lehobhob** [reply]
> Doesn't that need `http://` in front of it?

in Caddy to force http and avoid https redirection, yes. I have tried it with that and I still get http 502 error :(

**Reply by JC5** [reply]
I see you point to 8081, but the container usually runs at 8080.

**Reply by lehobhob** [reply]
Yes, that's because with Caddy, I have to enter the internal container port. Firefly is already on 8080, so I set the importer container to receive on port 8081, which proxies to 8080 inside the container.

**Reply by lehobhob** [reply]
> I see you point to 8081, but the container usually runs at 8080.

Actually you're right, it should have been the other way round, i need to enter the Caddy port on the right of the colon not the left. Unfortunately the port 8080 is already being used by firefly. I'll see if i can change that port

**Reply by lehobhob** [reply]
Is there an environment variable where I can set the container port of firefly data importer? 8080 is already being used by firefly itself, I think that might fix it

**Reply by JC5** [reply]
No, there isn't. It should not be a problem, not even for Caddy, that two containers both expose 8080. Since containers get their own IP space, this will not clash.

**Reply by lehobhob** [reply]
I thought so too but the error message I get with both ports on 8080 is:

Error response from daemon: driver failed programming external connectivity on endpoint firefly_iii_importer (container number): Bind for 0.0.0.0:8080 failed: port is already allocated

**Reply by hirusha-adi** [reply]
that is because you are exposing port `8080` on all interfaces (+ publicly) - `0.0.0.0:8080`. You don't need a `- ports` section at all. Instead, just comment it out and refer to it as follows within caddy: `firefly_iii:8080` (also make sure that both of them are on the same network)

**Comment by amca01** [comment]
> Does anyone have any experience with installing the Firefly III Data Importer with Docker Compose and Caddy? I've followed the instructions from the documentation (https://docs.firefly-iii.org/how-to/data-importer/installation/docker/)) but I'm getting a 502 error.
> 
> Firefly works fine but the data importer doesn't.
> 
> I have already set FIREFLY_III_URL in .importer.env to http://ffapp:8080
> 
> and VANITY_URL in .importer.env to http://localhost/
> 
> my caddy is the standard:
> 
> ```
> importer.{$MY_DOMAIN} {
> reverse_proxy firefly_iii_importer:8081
> }
> ```
> 
> I'm out of ideas of where it breaks :(

I appreciate this is a very old question, but I'm just starting to explore FireFly III - and I can't get it to work with Caddy.  I've set up an A record with my domain name provider; I'm using `firefly.mysite.net`.  My firefly core container uses ports 8082:8080 (as 8082 is not used by any other container.)  So my Caddyfile block is

firefly.mysite.net {
   reverse_proxy firefly_ii_core:8082
}

However, I'm getting 502 errors.  I just wondering if you might be able to toss any crumbs of your wisdom in my direction?  Many thanks!
